hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/command/index/TestSecondaryIndex.scala

```diff
@@ -400,6 +402,37 @@ class TestSecondaryIndex extends HoodieSparkSqlTestBase {
         // rider field (secondary key) should point to previous value
         val secondaryKey = timeTravelDF.where(s"_row_key = '$recordKeyToUpdate'").select("rider").collect().head.getString(0)
         assertEquals(initialSecondaryKey, secondaryKey)
+
+        // Perform Deletes on Records and Validate Secondary Index
+        val deleteDf = spark.read.format("hudi").load(basePath).filter(s"_row_key in ('${updateKeys.mkString("','")}')")
+        // Get fileId for the delete record
+        val deleteFileId = deleteDf.select("_hoodie_file_name").collect().head.getString(0)
+        deleteDf.write.format("hudi")
+          .options(hudiOpts)
+          .option(OPERATION.key, DELETE_OPERATION_OPT_VAL)
+          .mode(SaveMode.Append)
+          .save(basePath)
+        // Verify secondary index for deletes
+        validateSecondaryIndex(basePath, tableName, updateKeys, hasDeleteKeys = true)
+        // Corrupt the data file that was written for the delete key in the first instant
+        val firstCommitMetadata = deserializeCommitMetadata(metaClient.reloadActiveTimeline().getInstantDetails(firstInstant).get())
+        val partitionToWriteStats = firstCommitMetadata.getPartitionToWriteStats.asScala.mapValues(_.asScala.toList)
+        // Find the path for the given fileId
+        val matchingPath: Option[String] = partitionToWriteStats.values.flatten
+          .find(_.getFileId == deleteFileId)
+          .map(_.getPath)
+        assertTrue(matchingPath.isDefined)
+        // Corrupt the data file
```

Comment:
we need java docs to explain why we corrupt the data file. it may not be apparent. 
I can push out an update.